src/main/java/mods/railcraft/common/plugins/forge/CraftingPlugin.java

```diff
@@ -206,7 +206,7 @@ public static ResourceLocation getNameFromOutput(ItemStack output) {
         if (input instanceof IForgeRegistryEntry) {
             return ((IForgeRegistryEntry) input).getRegistryName();
         } else if (input instanceof ItemStack) {
-            return ((ItemStack) input).getItem().getRegistryName();
+            return getNameFromOutput((ItemStack) input); // prevents duplicate names!
```

Comment:
This seems incomplete? What about the other optional input types? They can still generate non-unique names.

I think perhaps the correct solution is to merge the two functions. Find a resource location, then pass it through the MultiSet unique counter to generate a unique one.

Reply:
So converting all those elements to item stacks and make a name?

Reply:
No need to convert them to ItemStacks, that is redundant. You already have a ResourceLocation, just add it to the recipeNames Multiset and append "$count" to the end of the path.

Reply:
Done.